src/mgr/Mgr.cc

```diff
@@ -409,6 +409,7 @@ void Mgr::shutdown()
       monc->sub_unwant("log-info");
       monc->sub_unwant("mgrdigest");
       monc->sub_unwant("fsmap");
+      monc->sub_unwant("servicemap");
```

Comment:
can we rely on monitor's behavior which removes the subscribers when the connection is reset?

Reply:
Sorry, I don't have a clear answer to that:-( 
Instead of unsubscribing 'servicemap', are you suggesting we should getting rid of all existing **redundant** `sub_unwant`  calls?

Reply:
yes.. the subscriptions are ephemeral state that is attached to the MonSession, which is cleaned up when the connection is lost.  That's a requirement because any daemon can die at any time and we can't leave a mess behind.

Reply:
i guess the question is whether there is any point to any of these sub_unwant calls at all.  probably not.

Reply:
after a second thought, i think we'd better remove all existing `sub_unwant()` calls unless we want to call `Mgr::init(). `sub_unwant()` only helps if the subscription is not acked by monitor, and MonClient is still alive, in that case MonClient will stop asking monitor for the subscription. but once a subscription is added, we cannot "undo" this without disconnect from monitor. so, to remove a sub in MonClient is a no-op under most circumstances. i guess that's why we don't call `sub_unwant()` elsewhere.

in this case, it's `MgrStandby::shutdown()` who starts the shutdown process. it stops monc, mgrc, then stops `Mgr`. so when `Mgr::stop()` is executed, MonClient is already disconnected.

the only use case of call `sub_unwant()` i can think of is `rados_monitor_log()`.